array_analyzer/extract/txt_parser.py

```diff
@@ -38,34 +53,124 @@ def create_xml_dict(path_):
         xmlstr = ET.tostring(xml_data, encoding='utf-8', method='xml')
         doc = xmltodict.parse(xmlstr)
 
-    # layout of array
-    layout = doc['configuration']['well_configurations']['configuration']['array']['layout']
+    try:
+        # layout of array
+        layout = doc['configuration']['well_configurations']['configuration']['array']['layout']
+
+        # fiducials
+        fiduc = layout['marker']
+
+        # spot IDs
+        spots = doc['configuration']['well_configurations']['configuration']['array']['spots']['spot']
 
-    # fiducials
-    fiduc = layout['marker']
+        # replicates
+        repl = doc['configuration']['well_configurations']['configuration']['array']['spots']['multiplet']
 
-    # spot IDs
-    spots = doc['configuration']['well_configurations']['configuration']['array']['spots']['spot']
+        array_params = dict()
+        array_params['rows'] = int(layout['@rows'])
+        array_params['columns'] = int(layout['@cols'])
+        array_params['v_pitch'] = float(layout['@vspace'])
+        array_params['h_pitch'] = float(layout['@hspace'])
+        array_params['spot_width'] = float(layout['@expected_diameter'])
+        array_params['bg_offset'] = float(layout['@background_offset'])
+        array_params['bg_thickness'] = float(layout['@background_thickness'])
+        array_params['max_diam'] = float(layout['@max_diameter'])
+        array_params['min_diam'] = float(layout['@min_diameter'])
 
-    # replicates
-    repl = doc['configuration']['well_configurations']['configuration']['array']['spots']['multiplet']
+        # not present in .xml but manually added here:
+        array_params['pixel_size_scienion'] = 0.0049
```

Comment:
Should these values be located here or would it make sense to move them to constants?